### PR TITLE
parallel_find : ... -> (unit -> 'a option) -> 'a option

### PR DESCRIPTION
**Pull request by gasche**
This is proposed as a solution for #89.

### CODE REVIEW

**Comment by gasche** [comment]
The CI failures comes from my use of `Domain.recommended_domain_count ()` in my test. The CI uses 5.0~alpha, and I guess this function was not available then? Updating the CI to the recent beta release should fix the issue. (I could also fix the domain number.) In the meantime I think it should be possible to review the code anyway -- the test does pass on my machine.

File: test/test_parallel_find.ml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  )
+
+let test_search pool ~chunk_size =
+  match search_needle pool ~chunk_size with
+  | None -> assert false
+  | Some needle ->
+    assert (Array.exists ((=) needle) needles)
+
+let () =
+  let num_domains = Domain.recommended_domain_count () in
```
[the remark this answers is not in the document]
Comment (gasche) [reply test/test_parallel_find.ml:31]: Good point, will fix. Thanks!

[the remark this answers is not in the document]
Comment (gasche) [reply test/test_parallel_find.ml]: Done, I rebased the PR with a fix.

**Comment by Sudha247** [comment]
I propose we merge this before #92 is worked on, if there are no objections.

**Comment by ElectreAAS** [comment]
As a user of domainslib (who doesn't know anything about the internals and has no opinion on #91 & #92), I can attest that this is a wanted feature!

**Comment by Sudha247** [comment]
Thanks for sharing @ElectreAAS. While we may want to add a full fledged cancellation mechanism (for e.g. like the one in Eio) at a later point, this is a useful feature to have at present. The CI failure looks unrelated, and the branch works fine on my local setup. Merging.

**Pull request merged**